dissect/target/loaders/velociraptor.py

```diff
@@ -116,14 +121,26 @@ def detect(path: Path) -> bool:
 
     def map(self, target: Target) -> None:
         os_type, dirs = find_fs_directories(self.root)
-        if os_type == OperatingSystem.WINDOWS:
-            # Velociraptor doesn't have the correct filenames for the paths "$J" and "$Secure:$SDS"
-            map_dirs(
-                target,
-                dirs,
-                os_type,
-                usnjrnl_path="$Extend/$UsnJrnl%3A$J",
-                sds_path="$Secure%3A$SDS",
-            )
-        else:
-            map_dirs(target, dirs, os_type)
+
+        # Velociraptor URL encodes paths before storing these in a collection, this leads plugins not being able to find
+        # these paths. To circumvent this issue, for a zip file the path names are URL decoded before mapping into the
+        # VFS and for a directory the paths are URL encoded at lookup time.
+        map_dirs(
+            target,
+            dirs,
+            os_type,
+            dirfs=VelociraptorDirectoryFilesystem,
+            zipfs=VelociraptorZipFilesystem,
+        )
+
+
+class VelociraptorDirectoryFilesystem(DirectoryFilesystem):
+    def _resolve_path(self, path: str) -> Path:
+        path = Path(quote(path, safe="$/"))
+        path = str(path.with_name(path.name.replace(".", "%2E", 1))) if path.name.startswith(".") else str(path)
```

Comment:
Using Path may not work exactly as intended on different platforms. I think in this case it's safer to use the dirname/basename/join helpers from fsutil.

Reply:
Fixed.

I improved the function due to files that have encoded values by default. For example, the EVTX file `Microsoft-Windows-Windows Defender%4WHC.evtx` is stored as `Microsoft-Windows-Windows Defender%254WHC.evtx` in a Velociraptor collection, thus the values `%<space>` are not encoded.